environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentCreationService.java

```diff
@@ -140,22 +140,36 @@ private Environment initializeEnvironment(EnvironmentCreationDto creationDto) {
     }
 
     private void initializeEnvironmentTunnel(Environment environment) {
-        Tunnel tunnel = environment.getExperimentalFeaturesJson().getTunnel();
-        boolean ccmV2Enabled = entitlementService.ccmV2Enabled(environment.getAccountId());
-        boolean overrideTunnel = environment.getExperimentalFeaturesJson().isOverrideTunnel();
+        ExperimentalFeatures experimentalFeatures = environment.getExperimentalFeaturesJson();
+        Tunnel tunnel = experimentalFeatures.getTunnel();
+        boolean overrideTunnel = experimentalFeatures.isOverrideTunnel();
         if (!overrideTunnel) {
-            if ((Tunnel.CCMV2 == tunnel || Tunnel.CCMV2_JUMPGATE == tunnel) && !ccmV2Enabled) {
-                throw new BadRequestException("CCMV2 not enabled for account.");
-            } else if (Tunnel.CCM == tunnel && ccmV2Enabled) {
-                ExperimentalFeatures experimentalFeaturesJson = environment.getExperimentalFeaturesJson();
-                experimentalFeaturesJson.setTunnel(Tunnel.CCMV2);
-                environment.setExperimentalFeaturesJson(experimentalFeaturesJson);
-                LOGGER.info("Environment is initialized with CCMV2 tunnel.");
+            boolean ccmV2Enabled = entitlementService.ccmV2Enabled(environment.getAccountId());
+            boolean ccmV2JumpgateEnabled = entitlementService.ccmV2JumpgateEnabled(environment.getAccountId());
+            checkCcmEntitlements(tunnel, ccmV2Enabled, ccmV2JumpgateEnabled);
+            if (Tunnel.CCM == tunnel) {
+                if (ccmV2JumpgateEnabled) {
+                    tunnel = Tunnel.CCMV2_JUMPGATE;
+                } else if (ccmV2Enabled) {
+                    tunnel = Tunnel.CCMV2;
+                }
+                experimentalFeatures.setTunnel(tunnel);
+                environment.setExperimentalFeaturesJson(experimentalFeatures);
             }
+            // TODO: if we want the Jumpgate to be default then this logic needs to be added for CCMV2 as well
         }
         LOGGER.info("Environment is initialized with [{}] tunnel.", tunnel);
     }
 
+    private void checkCcmEntitlements(Tunnel tunnel, boolean ccmV2Enabled, boolean ccmV2JumpgateEnabled) {
+        if (Tunnel.CCMV2_JUMPGATE == tunnel && !ccmV2JumpgateEnabled) {
```

Comment:
`ccmV2JumpgateEnabled` requires `ccmV2Enabled` to be true or are these totally different, and a customer can have jumpgate without the `ccmV2Enabled` entitlement?

Reply:
ccmV2JumpgateEnabled is a standalone entitlement and if it is set on customer account it will work